src/result-list.js

```diff
@@ -1,63 +1,65 @@
 'use strict';
 
-var ResponseHandler = require('./response-handler');
-var retrieve = require('./retrieve');
-
+import ResponseHandler from './response-handler';
+import retrieve from './retrieve';
+
+class ResultList {
+  constructor(search, options, onSuccess, onFailure) {
+    this._search = search;
+    this._options = options;
+    this._onSuccess = onSuccess;
+    this._onFailure = onFailure;
+    this._requestPromises = [];
+
+    // first call of next will increment to desired page
+    this._options.page -= 1;
+  }
 
-function ResultList(search, options, onSuccess, onFailure) {
-  this._search = search;
-  this._options = options;
-  this._onSuccess = onSuccess;
-  this._onFailure = onFailure;
-  this._requestPromises = [];
+  next() {
+    let nextPage = this._options.page += 1;
 
-  // first call of next will increment to desired page
-  this._options.page -= 1;
-}
+    if (this._requestPromises[nextPage]) {
+      return this._requestPromises[nextPage];
+    }
 
-ResultList.prototype.next = function() {
-  var nextPage = this._options.page += 1;
-
-  if (this._requestPromises[nextPage]) {
-    return this._requestPromises[nextPage];
+    return this._get();
   }
 
-  return this._get();
-};
+  previous() {
+    if (this._options.page <= 1) {
+      throw new Error('There is no previous page');
+    }
 
-ResultList.prototype.previous = function() {
-  if (this._options.page <= 1) {
-    throw new Error('There is no previous page');
-  }
+    let previousPage = this._options.page -= 1;
 
-  var previousPage = this._options.page -= 1;
+    if (this._requestPromises[previousPage]) {
+      return this._requestPromises[previousPage];
+    }
 
-  if (this._requestPromises[previousPage]) {
-    return this._requestPromises[previousPage];
+    return this._get();
   }
 
-  return this._get();
-};
+  _get() {
+    let page = this._options.page;
+    let perPage = this._options.per_page;
+    let promise =  retrieve(this._search, this._options)
+      .then(this._success.bind(this, page, perPage))
+      .fail(this._failure.bind(this, page, perPage));
 
-ResultList.prototype._get = function() {
-  var page = this._options.page;
-  var perPage = this._options.per_page;
-  var promise =  retrieve(this._search, this._options)
-    .then(this._success.bind(this, page, perPage))
-    .fail(this._failure.bind(this, page, perPage));
+    this._requestPromises[page] = promise;
+    return promise;
+  }
 
-  this._requestPromises[page] = promise;
-  return promise;
-};
+  _success(page, perPage, res) {
```

Comment:
You're right, I was confused. Since arrow functions are bound lexically, I think you can define one in `_success` every time you need to use `this`. Then there is no reason to `func.bind()` when you use it.

Source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions